docs/standard/exceptions/best-practices-for-exceptions.md

```diff
@@ -17,96 +17,158 @@ author: "mairaw"
-  
--   Use grammatically correct error messages, including ending punctuation. Each sentence in a description string of an exception should end in a period. For example, "The log table has overflowed.” would be an appropriate description string.  
-  
--   Provide <xref:System.Exception> properties for programmatic access. Provide additional properties for an exception (in addition to the description string) only when there's a programmatic scenario where the additional information is useful. For example, the <xref:System.IO.FileNotFoundException> provides the <xref:System.IO.FileNotFoundException.FileName%2A> property.  
-  
--   The stack trace begins at the statement where the exception is thrown and ends at the `catch` statement that catches the exception. Be aware of this fact when deciding where to place a `throw` statement.  
-  
--   Use exception builder methods. It is common for a class to throw the same exception from different places in its implementation. To avoid excessive code, use helper methods that create the exception and return it. For example:  
-  
-     [!code-cpp[Conceptual.Exception.Handling#6](../../../samples/snippets/cpp/VS_Snippets_CLR/conceptual.exception.handling/cpp/source.cpp#6)]
-     [!code-csharp[Conceptual.Exception.Handling#6](../../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.exception.handling/cs/source.cs#6)]
-     [!code-vb[Conceptual.Exception.Handling#6](../../../samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.exception.handling/vb/source.vb#6)]  
-  
-     Alternatively, use the exception's constructor to build the exception. This is more appropriate for global exception classes such as <xref:System.ArgumentException>.  
-  
--   Clean up intermediate results when throwing an exception. Callers should be able to assume that there are no side effects when an exception is thrown from a method.  
-  
+# Best practices for exceptions
+
+A well-designed app handles exceptions and errors to prevent app crashes. This section describes best practices for handling and creating exceptions.
+
+## Use try/catch/finally blocks
+
+Use `try`/`catch`/`finally` blocks around code that can potentially generate an exception. 
+
+In `catch` blocks, always order exceptions from the most specific to the least specific.
+
+Use a `finally` block to clean up resources, whether you can recover or not.
+
+## Handle common conditions without throwing exceptions
+
+For conditions that are likely to occur but might trigger an exception, consider handling them in a way that will avoid the exception. For example, if you try to close a connection that is already closed, you'll get an `InvalidOperationException`. You can avoid that by using an `if` statement to check the connection state before trying to close it.
```

Comment:
@tdykstra I have a feeling that we need to punt 🏈 on the nits right now ... e.g. from ...

> For example, if you ...

... to ...

> For example if you

... or just ...

> If you ...

Let's skip the nits on these passes. There's so much to do here that a plan will be needed to kick these bad boys into shape. I'm going to skip nits that won't haunt :ghost: me tonight! lol